### PR TITLE
Add linter for golangci.yml | fix linter errors in golangci.yml

### PR DESCRIPTION
**Pull request by dopey**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
```
Comment (azazeal) [inline .github/workflows/ci.yml:4]: Maybe on PR too?

Comment (azazeal) [reply .github/workflows/ci.yml:4]: This needs to run on PR too, @dopey

**Review by azazeal — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  pull_request:
+    types:
+      - opened
+      - synchronize
```
Comment (maraino) [inline .github/workflows/ci.yml:8]: We can add `reopened`, those 3 are the default values.

**Review by maraino — APPROVED** [review]
Consider to add reopened

**Review by maraino — APPROVED** [review]

**Pull request merged**